test/cmd/builds.sh

```diff
@@ -12,16 +12,64 @@ os::log::install_errexit
 url=":${API_PORT:-8443}"
 project="$(oc project -q)"
 
+# expect_build_output_spec runs `oc new-build` with given arguments and expects
+# a build config with a given name to be created with a given build output spec
+function os::cmd::builds::expect_build_output_spec() {
```

Comment:
```sh
template='{{with .spec.output.to}}{{.kind}} {{.name}}{{end}}'
os::cmd::expect_success 'oc new-build <args>'
os::cmd::expect_success_and_text "oc get bc/<name> --template '${template}'" '<text>'
```

This is fairly concise, clear, and doesn't have drawbacks for jUnit. I think it's OK if our tests are verbose, literal, and the less bash trickery we use the better.